### DISCUSSION TITLE
Runtime is very long

### DISCUSSION THREAD

**Discussion by vsolskyyy** [python/typing#1888] [General]
Hello, when running my program with mypy, runtime can be very long. My team has to run the program a lot, so I was wondering if anyone else was running into this issue, and if anybody has any tips. Thank you

**Comment by erictraut** [comment]
Long analysis time can be caused by many factors. Some of the more common causes are: a large number of source files in the project, files with many lines, a large number of overloads, large union types, and deeply-nested expressions.

Also, different type checkers have different optimizations. [Pyright](https://github.com/Microsoft/pyright) is often faster (sometimes significantly so) than mypy, but this is dependent on the code being analyzed.
